deps/rabbit/src/rabbit_peer_discovery.erl

```diff
@@ -108,29 +108,31 @@ maybe_init() ->
     %% node, even if the configuration changed in between.
     persistent_term:put(?PT_PEER_DISC_BACKEND, Backend),
 
-    _ = code:ensure_loaded(Backend),
-    case erlang:function_exported(Backend, init, 0) of
-        true  ->
-            ?LOG_DEBUG(
-               "Peer discovery: backend supports initialisation",
-               #{domain => ?RMQLOG_DOMAIN_PEER_DISC}),
-            case Backend:init() of
-                ok ->
-                    ?LOG_DEBUG(
-                       "Peer discovery: backend initialisation succeeded",
-                       #{domain => ?RMQLOG_DOMAIN_PEER_DISC}),
-                    ok;
-                {error, _Reason} = Error ->
-                    ?LOG_WARNING(
-                       "Peer discovery: backend initialisation failed: ~tp.",
-                       [Error],
-                       #{domain => ?RMQLOG_DOMAIN_PEER_DISC}),
-                    ok
-            end;
-        false ->
+    try
+        case Backend:init() of
+            ok ->
+                ?LOG_DEBUG(
+                   "Peer discovery: backend initialisation succeeded",
+                   #{domain => ?RMQLOG_DOMAIN_PEER_DISC}),
+                ok;
+            {error, _Reason} = Error ->
+                ?LOG_WARNING(
+                   "Peer discovery: backend initialisation failed: ~tp.",
+                   [Error],
+                   #{domain => ?RMQLOG_DOMAIN_PEER_DISC}),
+                ok
+        end
+    catch
+        error:undef ->
             ?LOG_DEBUG(
                "Peer discovery: backend does not support initialisation",
                #{domain => ?RMQLOG_DOMAIN_PEER_DISC}),
+            ok;
+        _:Reason:Stacktrace ->
+            ?LOG_ERROR(
+               "Peer discovery: backend initialisation failed: ~tp, ~tp",
+               [Reason, Stacktrace],
+               #{domain => ?RMQLOG_DOMAIN_PEER_DISC}),
             ok
```

Comment:
This comment might be for another future patch, but I’m writing it regardless :-)

In the spirit of not letting a node boot as standalone or from two separate clusters, should we continue to use the backend uninitialized?

The same is true if the function returns `{error, ...}`.